### ISSUE TITLE
 error in train_embedding\main.py evaluate function~

### ISSUE THREAD

**Issue by albert-jin** [malllabiisc/EmbedKGQA#89]
![image](https://user-images.githubusercontent.com/55691548/126035575-791ef3b1-c761-43b8-9eef-77e543f75533.png)

evaluate when  trained some epoches , the evaluation functions has error , next please see my debug infos

**Comment by albert-jin** [comment]
![image](https://user-images.githubusercontent.com/55691548/126035650-9c7025d9-8041-4514-8b2b-a0bdf4046b2a.png)
 
see the red circles, you can observe the variables er_vocab and er_vocab_pairs in the two bottom circles.

**Comment by albert-jin** [comment]
so when you get er_vocab[pair] you will always get [], which is  empty list , this may result the targets has no 1, the return value targets always return torch.zeros,  it`s no number 1 in targets~~

**Comment by albert-jin** [comment]
I wonder if you happend this condition like me. so need to refactor this function`s logits

**Comment by albert-jin** [comment]
I `m in China shanghai ,welcome to shanghai for traveling~  I will give my push request for you soon.

**Comment by albert-jin** [comment]
![image](https://user-images.githubusercontent.com/55691548/126036385-8e6905ad-0a8c-47a4-a3b1-486be0cddf1f.png)

thanks to this code, targets is not used in eval step, but this code is strange, refactor it if you want to enchance, everything depends by you....

**Comment by albert-jin** [comment]
if you have seen it, you can close it. @apoorvumang  thanks!

**Comment by apoorvumang** [comment]
it would be great if you could create a pull request with the refactored code, closing the issue for now
